src/arch/x86_64/kernel/apic.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(unaligned_references)]
+
```

Comment:
We definitely should't disable this lint.

Fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced).

I submitted a suggestion for the offending part of the code.

```suggestion
```